pkg/utils/kubernetes/object.go

```diff
@@ -87,17 +88,57 @@ func DeleteObjectsFromListConditionally(ctx context.Context, c client.Client, li
 	return flow.Parallel(fns...)(ctx)
 }
 
-// ResourcesExist checks if there is at least one object of the given gvk. The kind in the gvk must be the list kind,
-// for example corev1.SchemeGroupVersion.WithKind("PodList").
-func ResourcesExist(ctx context.Context, reader client.Reader, gvk schema.GroupVersionKind, listOpts ...client.ListOption) (bool, error) {
-	objects := &metav1.PartialObjectMetadataList{}
-	objects.SetGroupVersionKind(gvk)
+// ResourcesExist checks if there is at least one object of the given objList.
+func ResourcesExist(ctx context.Context, reader client.Reader, objList client.ObjectList, scheme *runtime.Scheme, listOpts ...client.ListOption) (bool, error) {
+	objects := objList
+
+	// Use `PartialObjectMetadata` if no or metadata only field selectors are passed (informer's indexers only have access to metadata fields).
+	if hasNoOrMetadataOnlyFieldSelector(listOpts...) {
+		gvk, err := apiutil.GVKForObject(objList, scheme)
+		if err != nil {
+			return false, err
+		}
+
+		objects = &metav1.PartialObjectMetadataList{}
+		(objects.(*metav1.PartialObjectMetadataList)).SetGroupVersionKind(gvk)
+	}
 
 	if err := reader.List(ctx, objects, append(listOpts, client.Limit(1))...); err != nil {
 		return true, err
 	}
 
-	return len(objects.Items) > 0, nil
+	switch o := objects.(type) {
+	case *metav1.PartialObjectMetadataList:
+		return len(o.Items) > 0, nil
+	default:
+		var objFound bool
+		if err := meta.EachListItem(objList, func(_ runtime.Object) error {
+			objFound = true
+			return nil
+		}); err != nil {
+			return false, err
+		}
+		return objFound, nil
```

Comment:
WDYT about using `ExtractList` ?
```suggestion
		list, err := meta.ExtractList(objects)
		if err == nil && len(list) > 0 {
			return true, nil
		}
		return false, err
```